### DISCUSSION TITLE
Lags when moving the canvas

### DISCUSSION THREAD

**Discussion by elmanAliev** [xyflow/xyflow#3382] [Q&A]
https://i.postimg.cc/T1X2TnHG/111.jpg

if there are a large number of nodes (I have custom nodes), and try to move the canvas, then lags occur.

the screenshot shows that at the time of moving the canvas, the Layerize process takes 16ms. is this normal? is it possible to improve this indicator so that there are no lags and loss of frames when moving the canvas?

**Comment by bcakmakoglu** [comment]
Screenshot link doesn't work (at least for me).

How many nodes are you trying to display? React Flow is not meant to handle thousands of nodes, so if that's your use-case you might want to consider using something else that's canvas based or sth.

**Comment by elmanAliev** [comment]
I updated the link.
hundreds of blocks are used. not thousands.

lags can occur when 200-300 blocks are used.

**Reply by bcakmakoglu** [reply]
Really depends on the complexity of your custom nodes though and what you're doing whenever the pane/nodes are moved.
Might be you're using very inefficient `useEffect` hooks or sth or your custom nodes themselves keep getting re-rendered causing them to re-compute stuff that causes delays. Hard to tell from the info you provided :/

**Comment by elmanAliev** [comment]
I am using react-flow-renderer 10.3.17.
all nodes are rendered only at the first rendering.
when moving the canvas, no functions are called, there are no listeners.

at the same time, the custom node has a complex structure.

should the Layerize process take so long (16ms) under such conditions

P. s. if I implement a simple structure for a node, then there are no lags, moving the canvas works smoothly.
